functions/go/set-labels/transformer/labels.go

```diff
@@ -0,0 +1,167 @@
+package transformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+type FieldPath []string
+type GVK struct {
+	group   string
+	version string
+	kind    string
+}
+
+// FieldSpec stores information about how to modify a specific label
+type FieldSpec struct {
+	Gvk       GVK
+	FieldPath FieldPath
+	// TODO: should support user configurable field for labels
+	CreateIfNotPresent bool
+}
+
+type FieldSpecs []FieldSpec
+
+// LabelTransformer stores information during the transform label process
```

Comment:
this struct does the actual "transform" operation. Can we change this to something like providing the main workflow ... it uses Config to parse functionConfig... it uses transform to change the labels ...